### ISSUE TITLE
Add support for .tld in @include

### ISSUE THREAD

**Issue by AuroraWright** [quoid/userscripts#589]
Some scripts use ".tld" in @include (as an example https://gist.github.com/ner00/ec9ae47e202b8e99f19be44a5af6baf3/raw/viewimage.user.js ) to match all possible tlds. Right now, these need to be modified to work with userscripts.

**Comment by AuroraWright** [comment]
Thanks for your reply and consideration, I don't want to push it further but I should maybe note that Greasemonkey does support it:

https://github.com/greasemonkey/greasemonkey/blob/f3e32fc506811e6956ad43b078667b0a3df62653/third-party/convert2RegExp.js

https://wiki.greasespot.net/Magic_TLD

So right now, scripts like the one above do actually work with Greasemonkey

**Comment by AuroraWright** [comment]
I just tested with the latest firefox and greasemonkey on macOS and it does work (the code is still in the repo, so I would have been surprised otherwise). I don't know if they are deprecating it (here you can see the script above being loaded and working)
<img width="593" alt="2" src="https://github.com/quoid/userscripts/assets/15571693/0b70ef8b-2a6f-4da8-b6cf-659c7013dd55">

<img width="1392" alt="3" src="https://github.com/quoid/userscripts/assets/15571693/01b8806d-abb1-46f5-bc6f-3e68463810b6">

**Comment by AuroraWright** [comment]
Also, this was broken on 4.1 and they fixed it in 4.2, so it doesn't look like they're planning to remove it https://github.com/greasemonkey/greasemonkey/issues/2795

**Comment by AuroraWright** [comment]
That's true, but it doesn't seem like it's going to be removed either if they're planning to improve it (for what it's worth I checked now and it seems to be implemented on Violentmonkey and Tampermonkey as well)

**Comment by quoid** [comment]
I haven't used `.tld` for `@include` before. Could either of you remind of the practical difference between `http*://*.foo.tld/` and `http*://*.foo.*/`?

**Comment by quoid** [comment]
@ACTCD Thanks for clarifying; it seems to be a pure convenience "feature".  I reckon if an author wanted they could create a more restrictive regex pattern, for `@include`, that has the convenience that `.tld` brings 🧐.

**Comment by AuroraWright** [comment]
I think you raise a valid point, from a developer point of view. I’m honestly not a userscript developer but I’d follow your advice if i were one. The issue is still very much there from a user point of view though, imo. Some scripts on greasyfork etc (not sure about the amount, it’d be interesting to know) do use .tld and right now they just silently fail here so an user would just blame the extension and consider switching to the (closed source, potentially worse for privacy/security) Tampermonkey. Even if an user is also knowledgeable about userscripts, changing this every time there’s an update is really annoying especially on ios (which most people are using this on, probably). Maybe it’d make sense to make this an off by default, toggleable compatibility feature?
